native/src/main/java/io/ballerina/stdlib/websub/NativeHttpToWebsubAdaptor.java

```diff
@@ -103,8 +104,22 @@ private static Object invokeRemoteFunction(Environment env, BObject bSubscriberS
         StrandMetadata metadata = new StrandMetadata(module.getOrg(), module.getName(), module.getVersion(),
                 parentFunctionName);
         Object[] args = new Object[]{message, true};
-        env.getRuntime().invokeMethodAsync(bSubscriberService, remoteFunctionName, null, metadata,
-                new SubscriberCallback(balFuture, module), args);
+
+        if (bSubscriberService.getType().isIsolated()) {
```

Comment:
Shall we combine the conditions in L#108 and L#114?

Reply:
Here we are checking two conditions.
- L#108: Whether whole service object is isolated; then no need to check whether methods are isolated
- L#114: Even though service object is not isolated whether the particular method is isolated

Hence, do we need to combine them ?

Reply:
Resolved with [2aff8c5d1be1d3c4c079ebcea79a4840cefa075d](https://github.com/ballerina-platform/module-ballerina-websub/pull/421/commits/2aff8c5d1be1d3c4c079ebcea79a4840cefa075d)